File: aoc2022/day4.py
import re
with open('input/day4.in') as f:
    nums = [list(map(int, re.sub('\D+', " " ,l.strip()).split(' '))) for l in f]
contains = len(list(filter (lambda x: x == True, (list(map(lambda x: (x[0] <= x[2] and x[1] >= x[3]), nums))))))
print(contains)

File: aoc2022/day4.hs
import Functions

main = do
  input <- map parseList . lines <$> readFile "input/day4.in"
  let ansP1 = length . filter id $ map (\[x, y, z, w] -> x <= z && y >= w || x >= z && y <= w) input
      ansP2 = length . filter id $ map (\[x, y, z, w] -> y >= z && w >= x) input
  print ansP1
  print ansP2
